pkg/github/actions/runnerscaleset.go

```diff
@@ -38,7 +39,11 @@ func (client *ActionsClient) CreateRunnerScaleSet(ctx context.Context, runner *t
 func (client *ActionsClient) DeleteRunnerScaleSet(ctx context.Context, runnerScaleSetId int) error {
 	path := fmt.Sprintf("/%s/%d", scaleSetEndpoint, runnerScaleSetId)
 
-	_, err := RequestJSON[any, any](ctx, client, http.MethodDelete, path, nil)
+	operation := func() error {
+		_, err := RequestJSON[any, any](ctx, client, http.MethodDelete, path, nil)
 
-	return err
+		return err
+	}
+
+	return backoff.Retry(operation, backoff.NewExponentialBackOff())
```

Comment:
Looking at how and where we call `DeleteRunnerScaleSet`, I do not think this is needed.

Why?

We call it in two places:
1. When terminating the application gracefully.
2. When starting the application and the scaleset exists (which means (1) did not complete)

A backoff retry on termination is either going to delay the termination with (max) 15 min or be cancelled at some point.
Even if the deletion fails on termination, starting the integration again will delete it. If the deletion fails, it will fail fast, which is not a bad thing. A retry backoff could delay the error with up to 15 min.

So my suggestion is to remove it so we do not affect the startup or the shutdown of the application.

Reply:
This was removed